### ISSUE TITLE
Add compatibility with PHP 8.0

### ISSUE THREAD

**Issue by amotl** [crate/crate-dbal#105]
Hi there,

PHP 8.0.1 has been released and PHP 8.1.x is in the making. We might want to follow that.

With kind regards,
Andreas.

**Comment by amotl** [comment]
Now that [`crate-pdo`](https://github.com/crate/crate-pdo) version 2.0.0 has been released, we can look into adding PHP8 support here.

**Comment by amotl** [comment]
Thanks for working on #111, @alexander-schranz. [1] gives some first insights into what has to be made compatible.

```php
Argument 1 passed to Doctrine\DBAL\Connection::ensureForwardCompatibilityStatement() must implement interface Doctrine\DBAL\Driver\ResultStatement, instance of Crate\PDO\PDOStatement given, called in /home/runner/work/crate-dbal/crate-dbal/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php on line 1314
```

[1] https://github.com/crate/crate-dbal/runs/2452765877#step:7:17

**Comment by amotl** [comment]
Same error on #113 with PHP7.4, see [2]. So, this is not specific to PHP8, right?

[2] https://github.com/crate/crate-dbal/runs/2453058434#step:7:17

**Comment by alexander-schranz** [comment]
Looks for me like it is still a `Crate\PDO\PDOStatement` instead of our new set `Doctrine\DBAL\Driver\PDO\Statement`. Not sure what I missed here.

**Comment by amotl** [comment]
Resolved through #111 by @alexander-schranz. Thanks a stack.
